addons/docs/src/frameworks/vue/sourceDecorator.ts

```diff
@@ -0,0 +1,198 @@
+/* eslint no-underscore-dangle: ["error", { "allow": ["_vnode"] }] */
+
+import { addons, StoryContext } from '@storybook/addons';
+import { logger } from '@storybook/client-logger';
+import prettier from 'prettier/standalone';
+import prettierHtml from 'prettier/parser-html';
+import Vue from 'vue';
+
+import { SourceType, SNIPPET_RENDERED } from '../../shared';
+
+export const skipJsxRender = (context: StoryContext) => {
+  const sourceParams = context?.parameters.docs?.source;
+  const isArgsStory = context?.parameters.__isArgsStory;
+
+  // always render if the user forces it
+  if (sourceParams?.type === SourceType.DYNAMIC) {
+    return false;
+  }
+
+  // never render if the user is forcing the block to render code, or
+  // if the user provides code, or if it's not an args story.
+  return !isArgsStory || sourceParams?.code || sourceParams?.type === SourceType.CODE;
+};
+
+export const sourceDecorator = (storyFn: any, context: StoryContext) => {
```

Comment:
Named this `sourceDecorator` instead of `jsxDecorator` because it's not a JSX... if it's a requirement I'll change this.